packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -133,36 +110,37 @@ export class OpenID4VCIClient {
     }
 
     // add 'openid' scope if not present
-    if (scope && !scope.includes('openid')) {
-      scope = `openid ${scope}`;
+    if (!scope?.includes('openid')) {
+      scope = ['openid', scope].filter((s) => !!s).join(' ');
     }
 
-    //fixme: handle this for v11
-    const queryObj = {
+    const queryObj: { [key: string]: string } = {
       response_type: ResponseType.AUTH_CODE,
-      client_id: clientId,
+      client_id: this.clientId || '',
       code_challenge_method: codeChallengeMethod,
       code_challenge: codeChallenge,
       authorization_details: JSON.stringify(this.handleAuthorizationDetails(authorizationDetails)),
       redirect_uri: redirectUri,
       scope: scope,
-    } as AuthorizationRequestV1_0_09;
+    };
+
+    if (this.credentialOffer.issuerState) {
+      queryObj['issuer_state'] = this.credentialOffer.issuerState;
+    }
 
     return convertJsonToURI(queryObj, {
       baseUrl: this._endpointMetadata.authorization_endpoint,
-      uriTypeProperties: ['redirect_uri', 'scope', 'authorization_details'],
-      version: this.version(),
+      uriTypeProperties: ['redirect_uri', 'scope', 'authorization_details', 'issuer_state'],
```

Comment:
Will look later into whether leaving the version out doesn't have any side effects

Reply:
Based on the current `convertJsonToURI` implementation, excluding version param will do exactly what is needed in this case. Provided attributes have be added as query params. [else statement](https://github.com/Sphereon-Opensource/OID4VCI/blob/develop/packages/common/lib/functions/Encoding.ts#L36) does exactly that.

Reply:
Yeah, the problem however is that the convertJsonToURI is called from many places and it for sure needs the logic to distinguish between V11 and higher in other areas. What I will do is add an additional param that takes precedence over the version. From the first look of it you need it from the `formPost` method. So will use that param there